nautobot/core/templates/generic/object_retrieve.html

```diff
@@ -175,26 +175,28 @@ <h1>
                                 <div class="table-responsive">
                                     {% render_table associated_contacts_table 'inc/table.html' %}
                                 </div>
-                                <div class="panel-footer noprint">
-                                    {% if perms.extras.change_contactassociation %}
-                                        <button type="submit" name="_edit" formaction="{% url 'extras:contactassociation_bulk_edit' %}?return_url={{request.path}}" class="btn btn-warning btn-xs">
-                                            <span class="mdi mdi-pencil" aria-hidden="true"></span> Edit
-                                        </button>
-                                    {% endif %}
-                                    {% if perms.extras.delete_contactassociation %}
-                                        <button type="submit" formaction="{% url 'extras:contactassociation_bulk_delete' %}?return_url={{request.path}}" class="btn btn-danger btn-xs">
-                                            <span class="mdi mdi-trash-can-outline" aria-hidden="true"></span> Delete
-                                        </button>
-                                    {% endif %}
-                                    {% if perms.extras.add_contactassociation %}
-                                        <div class="pull-right">
-                                            <a href="{% url 'extras:object_contact_team_assign' %}?return_url={{request.path}}&associated_object_id={{object.id}}&associated_object_type={{content_type.id}}" class="btn btn-primary btn-xs">
-                                                <span class="mdi mdi-plus-thick" aria-hidden="true"></span> Add Contact
-                                            </a>
-                                        </div>
-                                    {% endif %}
-                                    <div class="clearfix"></div>
-                                </div>
+                                {% with request.path|add:"?tab=contacts" as return_url %}
```

Comment:
I think you need to urlencode `request.path` here. This is my url after clicking on Add Contact on a device type detail view: `http://localhost:8080/extras/contact-associations/assign-contact-team/?return_url=/dcim/device-types/e585e275-df36-4dca-9cc3-1601a6d1b690/?tab=contacts&associated_object_id=e585e275-df36-4dca-9cc3-1601a6d1b690&associated_object_type=59`

Notice the two `?` in the url